### ISSUE TITLE
Wrong version calculation when merging release via github pull request

### ISSUE THREAD

**Issue by SergeyBear** [GitTools/GitVersion#3983]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched [issues](https://github.com/GitTools/GitVersion/issues) to ensure it has not already been reported

### GitVersion package

GitVersion.Tool

### GitVersion version

5.10 - 6.0.0-beta.5

### Operating system

Linux

### What are you seeing?

Branches:
`master` (calculated `0.1.0`)
`release-0.2` (calculated `0.2.0-alpha.1`)

Merging `release-0.2` to `master` via github pull request:
`master` (calculated `0.1.0`)

Merging `release-0.2` to `master` via `git merge`:
`master` (calculated `0.2.0`)

### What is expected?

Merging `release-0.2` to `master` via github pull request:
`master` (calculated `0.2.0`)

### Steps to Reproduce

Init `master`, create `releas-0.2` branch and merge `release-0.2` to `master` via github pull request

**Comment by SergeyBear** [comment]
I found that merging via github pull request set the default commit title to:
`Merge pull request #1 from foo/release-0.2`
while merging via `git merge` sets the default commit title to:
`Merge branch 'release-0.2'`

When I set custom title to `Merge branch 'release-0.2'` when merging github pull request - the version is calculated currectly (0.2.0) and in gitversion log I see new messages, that source branch is found in merge commit message.

In past issues I found some solutions for AzureDevOps, based on description templates, but Github has no merge message templates (feature request is still pending) and adding custom title on every pull request is a pain. After some tests I found a quickfix:
```merge-message-formats:
  github: '^Merge pull request #(?<PullRequestNumber>\d+) from .+/(?<SourceBranch>.+)'
```
but I think that it is better to add this merge message format out of the box, so default github merge title worked without needing to configuring gitversion for new github users.
